components/Chips/src/MDCChipView.m

```diff
@@ -0,0 +1,556 @@
+    borderColor = _borderColors[@(UIControlStateNormal)];
+  }
+  return borderColor;
+}
+
+- (void)setBorderColor:(nullable UIColor *)borderColor forState:(UIControlState)state {
+  _borderColors[@(state)] = borderColor;
+
+  [self updateBorderColor];
+}
+
+- (void)updateBorderColor {
+  self.layer.borderColor = [self borderColorForState:self.state].CGColor;
+}
+
+- (CGFloat)borderWidthForState:(UIControlState)state {
+  NSNumber *borderWidth = _borderWidths[@(state)];
+  if (borderWidth == nil && state != UIControlStateNormal) {
+    borderWidth = _borderWidths[@(UIControlStateNormal)];
+  }
+  if (borderWidth != nil) {
+    return (CGFloat)borderWidth.doubleValue;
+  }
+  return 0;
+}
+
+- (void)setBorderWidth:(CGFloat)borderWidth forState:(UIControlState)state {
+  _borderWidths[@(state)] = @(borderWidth);
+
+  [self updateBorderWidth];
+}
+
+- (void)updateBorderWidth {
+  self.layer.borderWidth = [self borderWidthForState:self.state];
+}
+
+- (CGFloat)elevationForState:(UIControlState)state {
+  NSNumber *elevation = _elevations[@(state)];
+  if (elevation == nil && state != UIControlStateNormal) {
+    elevation = _elevations[@(UIControlStateNormal)];
+  }
+  if (elevation != nil) {
+    return (CGFloat)[elevation doubleValue];
+  }
+  return 0;
+}
+
+- (void)setElevation:(CGFloat)elevation forState:(UIControlState)state {
+  _elevations[@(state)] = @(elevation);
+
+  [self updateElevation];
+}
+
+- (void)updateElevation {
+  CGFloat newElevation = [self elevationForState:self.state];
+  if (self.layer.elevation != newElevation) {
+    self.layer.elevation = newElevation;
+  }
+}
+
+- (nullable UIColor *)titleColorForState:(UIControlState)state {
+  UIColor *titleColor = _titleColors[@(state)];
+  if (!titleColor && state != UIControlStateNormal) {
+    titleColor = _titleColors[@(UIControlStateNormal)];
+  }
+  return titleColor;
+}
+
+- (void)setTitleColor:(nullable UIColor *)titleColor forState:(UIControlState)state {
+  _titleColors[@(state)] = titleColor;
+
+  [self updateTitleColor];
+}
+
+- (void)updateTitleColor {
+  self.titleLabel.textColor = [self titleColorForState:self.state];
+}
+
+- (void)updateState {
+  [self updateBackgroundColor];
+  [self updateBorderColor];
+  [self updateBorderWidth];
+  [self updateElevation];
+  [self updateTitleColor];
+}
+
+#pragma mark - Control
+
+- (void)setEnabled:(BOOL)enabled {
+  [super setEnabled:enabled];
+
+  [self updateState];
+}
+
+- (void)setHighlighted:(BOOL)highlighted {
+  [super setHighlighted:highlighted];
+
+  [self updateState];
+}
+
+- (void)setSelected:(BOOL)selected {
+  [super setSelected:selected];
+
+  [self updateState];
+  [self setNeedsLayout];
+}
+
+#pragma mark - Layout
+
+- (void)layoutSubviews {
+  _inkView.frame = self.bounds;
+  _imageView.frame = [self imageViewFrame];
+  _selectedImageView.frame = [self selectedImageViewFrame];
+  _accessoryView.frame = [self accessoryViewFrame];
+  _titleLabel.frame = [self titleLabelFrame];
```

Comment:
Please use property accessors unless there's a need to access the backing iVar here.

Reply:
Why?